File: String/1358. Number of Substrings Containing All Three Characters.cpp
class Solution {
    public:
        int numberOfSubstrings(string s) 
        {
            int n = s.size(), ans = 0;
            vector<int> lastSeen(100);
            for(int i = 0; i <= n; i++)
            {
                ans += min({lastSeen['a'], lastSeen['b'], lastSeen['c']});
                lastSeen[s[i]] = i + 1;
            }    
            return ans;
        }
    };

File: 1358. Number of Substrings Containing All Three Characters.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        ans, n = 0, len(s)
        a, b, c = -1, -1, -1 
        for i in range(n):
            if s[i] == 'a': a = i
            elif s[i] == 'b': b = i
            elif s[i] == 'c': c = i
            ans += min(a, b, c) + 1
        return ans